### ISSUE TITLE
The top-level v-if causes an error on the v-else side if vue-loader#optimizeSSR is true

### ISSUE THREAD

**Issue by ooxif** [vuejs/vue-loader#898]
### Version
13.0.2

### Reproduction link
[https://github.com/ooxif/vue-ssr-v-if](https://github.com/ooxif/vue-ssr-v-if)

### Steps to reproduce
1. Clone this repo
2. `yarn install && yarn build && yarn start`
3. Visit `http://localhost:8080` - Error
4. Set `optimizeSSR` to `false` at `build/webpack.base.config.js`
5. `yarn build && yarn start`
6. Visit `http://localhost:8080` - OK

### What is expected?
Shows "v-else - Error if vue-loader#optimizeSSR is true"

### What is actually happening?
Results in a blank page, the console reports an error

---
Uncaught DOMException: Failed to execute 'appendChild' on 'Node': This node type does not support this method.
at Object.appendChild (http://localhost:8080/dist/vendor.dc2a8198a413dd9d9a70.js:6182:8)
at insert (http://localhost:8080/dist/vendor.dc2a8198a413dd9d9a70.js:6481:17)
at createElm (http://localhost:8080/dist/vendor.dc2a8198a413dd9d9a70.js:6409:7)
at createChildren (http://localhost:8080/dist/vendor.dc2a8198a413dd9d9a70.js:6489:9)
at hydrate (http://localhost:8080/dist/vendor.dc2a8198a413dd9d9a70.js:6778:11)
at VueComponent.patch [as __patch__] (http://localhost:8080/dist/vendor.dc2a8198a413dd9d9a70.js:6855:17)
at VueComponent.Vue._update (http://localhost:8080/dist/vendor.dc2a8198a413dd9d9a70.js:3688:19)
at VueComponent.updateComponent (http://localhost:8080/dist/vendor.dc2a8198a413dd9d9a70.js:3812:10)
at Watcher.get (http://localhost:8080/dist/vendor.dc2a8198a413dd9d9a70.js:4155:25)
at new Watcher (http://localhost:8080/dist/vendor.dc2a8198a413dd9d9a70.js:4144:12)

**Comment by aeharding** [comment]
Aha! I'm not going crazy. Thanks for making a reproduction! Saved me from a couple hours to get a bug report out there.

I also notice this in my app upgrading to Vue 2.4, and when I disable `optimizeSSR` it works again.

Thanks! ❤️

**Comment by alanwillms** [comment]
I have the same issue here, but when I turn off `optimizeSSR` instead of getting a console error my entire component simply disappears from DOM after hydrating:

```vue
<template>
  <div>
    <app-header>
      <h1>{{ $t('categories.action_index_name') }}</h1>
    </app-header>

    <div v-if="categories && categories.length > 0">
      <category-card
        v-for="category in categories"
        :key="category.id"
        :category="category">
      </category-card>
    </div>

    <div v-else>
      <category-placeholder></category-placeholder>
    </div>
  </div>
</template>
```

**Comment by Linzdigr** [comment]
Same issue with vue-loader 13.6.2. Any news about this?

**Comment by yyx990803** [comment]
This has been fixed in vue core (tested with 2.5.13)
